### ISSUE TITLE
Rule suggestion: symbol-description

### ISSUE THREAD

**Issue by jrencz** [eslint/eslint#6778]
_(bug report template was previously here. It was removed)_

I suggest new rule: `symbol-description` with options `always` and `never`: It should check if all (or neither of) symbols created in code have the description as described in the Spec: [19.4.1.1 Symbol ( [ description ] )](http://www.ecma-international.org/ecma-262/6.0/#sec-symbol-description)

Rationale: symbol description is optional but it facilitates debugging because [Symbol#toString](http://www.ecma-international.org/ecma-262/6.0/#sec-symbol.prototype.tostring) returns a [descriptive string](http://www.ecma-international.org/ecma-262/6.0/#sec-symboldescriptivestring) containing symbol description so it's worth enforcing.

**When does this rule warn? Please describe and show example code:**
The rule as briefly described above has 2 settings: `always` and `never`

With `always` setting it should warn if the following code is detected:

```js
Symbol();
```

With `never` setting it should warn if the following code is detected:

```js
Symbol('any strng');
Symbol(anyVariable);
```

---
EDIT:

As @nzakas [asked me to do](https://github.com/eslint/eslint/issues/6778#issuecomment-236673344) I'm updating the request with the rule proposal template:

**Is this rule preventing an error or is it stylistic?**

The rule neither prevents an error nor is stylistic. I'd say it belongs either to "Best Practices" or to "ECMAScript 6" groups.

**Why is this rule a candidate for inclusion instead of creating a custom rule?**

It covers one of the new language features. If it's in the core people may adopt good practice of describing symbols used in their code by analysing the rules list and finding this one.

**Are you willing to create the rule yourself?**

Yes, but I never implemented eslint rule so I'm going to need some help.

**Comment by mysticatea** [comment]
Thank you for this issue.

I like this idea :+1:.

**Comment by platinumazure** [comment]
My only question is whether this is important enough to meet our new rule criteria. If not, I could see a use for a symbol-themed plugin.

Not sure if I want to endorse just yet, but I'm not doing thumbs-down either.

**Comment by nzakas** [comment]
@jrencz please read over our [guidelines for submitting rule proposals](http://eslint.org/docs/developer-guide/contributing/new-rules) and update your description to match the format.

In general I like the idea, but I'd like to see a proper proposal before evaluating.

**Comment by jrencz** [comment]
@nzakas I updated the description

**Comment by platinumazure** [comment]
Now I understand what users need to do to conform to the rule (seems reasonable for symbols to have a description). I'll champion.

**Comment by jrencz** [comment]
@platinumazure I made some quick and dirty attempt to implement the rule as described: https://github.com/jrencz/eslint/tree/symbol-description

Should I file a PR as-is (with no documentation) or should I better prepare some docs first?

**Comment by platinumazure** [comment]
@jrencz Definitely add docs before submitting a PR :smile: But if you would like, I am happy to take a look at your WIP branch.

**Comment by jrencz** [comment]
@platinumazure I'd appreciate. I'll add documentation tomorrow.

I based my code on rules radix and no-new-symbol

**Comment by vitorbal** [comment]
I am :+1: for this rule proposal.

**Comment by mysticatea** [comment]
I'm not sure if we need the`never` option. I think no option is more simple.
That implementation looks good to me :wink:

**Comment by vitorbal** [comment]
@mysticatea agreed, I don't think the `never` option is of much use.

**Comment by nzakas** [comment]
Agreed, I don't think "never" is necessary. 

I'm 👍
